tests/test_decl.py

```diff
@@ -451,6 +451,99 @@ def test_delete_operations_delete_attributes_if_no_list_of_uuids_was_given(
         assert len(root_function.functions) == 0
 
 
+class TestApplyExtend:
+    @staticmethod
+    def test_extend_operations_change_model_objects_in_place(
+        model: capellambse.MelodyModel,
+    ) -> None:
+        root_component = model.by_uuid(ROOT_COMPONENT)
+        fnc = model.by_uuid(ROOT_FUNCTION)
+        in_functions_before_apply = fnc in root_component.functions
+        yml = f"""\
+            - parent: !uuid {ROOT_COMPONENT}
+              extend:
+                functions:
+                  - !uuid {fnc.uuid}
```

Comment:
Same as in the doc: Because components cannot "own" functions, this doesn't move the `fnc` but only creates a new reference to it. The best way to ensure that something was actually moved is checking whether its `.parent` changed:

```python
old_parent = fnc.parent
decl.apply(...)
assert old_parent != fnc.parent  # AssertionError
```

Please add this to the test. :)